packages/access-client/src/agent-data.js

```diff
@@ -171,12 +171,17 @@ export const isSessionProof = (delegation) =>
  * Get a map from CIDs to the session proofs that reference them
  *
  * @param {AgentData} data
+ * @param {Ucanto.DID} [issuer] - if provided, will only return session proofs issued by this id
  * @returns {Record<string, Ucanto.Delegation>}
  */
-export function getSessionProofs(data) {
+export function getSessionProofs(data, issuer) {
```

Comment:
Part of what was causing the bug is that this function ends up building a map of CIDs to proofs. The CID it uses is the nb.proof from a ucan/attest. The agent might actually have multiple ucan/attest ucans with the same nb.proof CID, but with different issuers. Previously only one of these delegations would end up in the returned map from this function, and the caller had no way of getting a session proof appropriate for the audience of their request.

Reply:
I think this is fairly low level API so I suggest a breaking change that will change return type to 

```js
Record<Ucanto.DID, Record<string, ...Ucanto.Delegation>>
```

That way things will work correctly even if you do not pass the issuer. Furthermore I do have use case where I query `proofs` and I do not know the issuer of attestations so getting all the proofs would be desired there.

Reply:
I did something like that here https://github.com/web3-storage/w3up/pull/1047/commits/a9aedf67d51b66aeb9a3653237565f67565ade61